### PR TITLE
Remove la obr except 54089 8

### PR DESCRIPTION
**Pull request by somesylvie**
# Description

- Add LA OBR removal Transformation
  - This is for the removal of all extra OBRs that are not '54089-8'
- Add LA Sample files for testing
- Add RS e2e testing with new sample file and assertions
  - This was tested locally with a local TI and RS instance. If you test this functionality, please run this locally. 
  - Note: This RS e2e test does not work currently because the transformation code is not in staging.

## Issue
https://github.com/CDCgov/trusted-intermediary/issues/1355

## Checklist

- [x] I have added tests to cover my changes

### CODE REVIEW

File: etor/src/main/resources/transformation_definitions.json [hunk 1]
```diff
@@ -167,6 +167,24 @@
         }
       ]
     },
+    {
+      "name": "laOchsnerOruRemoveObservationRequests",
+      "description": "Removes all OBRs from an LA Ochsner ORU message except for the OBR with value '54089-8' in OBR-4.1. All OBXs are attached to the sole remaining OBR",
+      "message": "",
+      "conditions": [
+        "Bundle.entry.resource.ofType(MessageHeader).destination.extension.where(url = 'https://reportstream.cdc.gov/fhir/StructureDefinition/universal-id').value = '1.2.840.114350.1.13.286.2.7.2.695071'",
```
Comment (somesylvie) [inline etor/src/main/resources/transformation_definitions.json:175]: Once we're closer to go live/have better test data, we should confirm whether this value is environment-specific or not. @scleary1cs what's a good way to track this?

File: rs-e2e/src/main/resources/assertion_definitions.json [hunk 2]
```diff
@@ -101,6 +101,16 @@
         "OBX-3.1 = '57721-3'",
         "OBX.count() = 1"
       ]
+    },
+    {
+      "name": "LADPH ORU requirements",
```
Comment (GilmoreA6) [inline rs-e2e/src/main/resources/assertion_definitions.json:106]: nit: the name is not quite right. It's not LADPH that is requiring the transform, it is Ochsner. The name should reflect this and be more descriptive like "LA-Oschner OBR flattening"

File: etor/src/main/resources/transformation_definitions.json (hunk 1 above)
Comment (basiliskus) [reply etor/src/main/resources/transformation_definitions.json:175]: It is environment-specific. This is the value for staging. When adding the setting in production, we need to update this OID

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 [hunk 3]
```diff
@@ -0,0 +1,56 @@
+MSH|^~\&|Natus^natus.health.state.la.us^DNS|Louisiana Office of Public Health Laboratory^2.16.840.1.114222.4.3.26.3.1^ISO|Test Department of tests^1.2.840.114350.1.13.286.2.7.2.695071^ISO|^simulated-hospital-id^L,M,N|20240101050000+0000||ORU^R01^ORU_R01|20240101050000_0001|T|2.5.1|||AL|AL|||||LRI_NG_FRN_PROFILE^^2.16.840.1.113883.9.195.3.4^ISO~LRI_NDBS_COMPONENT^^2.16.840.1.113883.9.195.3.6^ISO~LAB_PRN_Component^^2.16.840.1.113883.9.81^ISO~LAB_TO_COMPONENT^^2.16.840.1.113883.9.22^ISO
```
Comment (basiliskus) [inline examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: If this file is is the resulting output for 003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7, we can run the `scripts/setup/update-examples-snapshots.sh` script to automatically create and update all the snapshots. That could be included in a separate PR

File: examples/LA/004_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_example_2.hl7 [hunk 4]
```diff
@@ -0,0 +1,56 @@
+MSH|^~\&|Natus^natus.health.state.la.us^DNS|Louisiana Office of Public Health Laboratory^2.16.840.1.114222.4.3.26.3.1^ISO|Test Department of tests^1.2.840.114350.1.13.286.2.7.2.695071^ISO||20240101011000+0000||ORU^R01^ORU_R01|20240101011000_0002|T|2.5.1|||AL|AL|||||LRI_NG_FRN_PROFILE^^2.16.840.1.113883.9.195.3.4^ISO~LRI_NDBS_COMPONENT^^2.16.840.1.113883.9.195.3.6^ISO~LAB_PRN_Component^^2.16.840.1.113883.9.81^ISO~LAB_TO_COMPONENT^^2.16.840.1.113883.9.22^ISO
```
Comment (basiliskus) [inline examples/LA/004_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_example_2.hl7:1]: Same comment as for the other file

File: rs-e2e/src/main/resources/assertion_definitions.json [hunk 5]
```diff
@@ -101,6 +101,16 @@
...
+      "conditions": [
+        "MSH-10 = '010'"
```
Comment (basiliskus) [inline rs-e2e/src/main/resources/assertion_definitions.json:108]: Instead of using the MSH-10 value, the conditions here could be:
```
"MSH-9.2 = 'R01'",
"MSH-5.2 = '2.16.840.1.114222.4.3.26.3.2'"
```
That means the assertion will be made for any LA ORU

Comment (saquino0827) [reply rs-e2e/src/main/resources/assertion_definitions.json:108]: I mainly used `MSH-10 = '010'` because I like how it targets that specific sample file. I don't know how I feel about mixing and matching different test files and conditions for assertions. Maybe this can change once we have a better idea of all of the LA transformation.

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: What does updating snapshots do? Not sure I've ever used the script

File: rs-e2e/src/main/resources/assertion_definitions.json (hunk 5 above)
Comment (basiliskus) [reply rs-e2e/src/main/resources/assertion_definitions.json:108]: Sounds good

File: etor/src/main/resources/transformation_definitions.json (hunk 1 above)
Comment (saquino0827) [reply etor/src/main/resources/transformation_definitions.json:175]: I'll write up a card for this and post it here

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (basiliskus) [reply examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: It's a new script. It iterates over all the files with the `_0_initial_message.hl7` suffix in the `examples/` folder, sends them to a local instance of RS, and then downloads the snapshots at different points of the flow that we care about, including the final delivery. The idea is to keep all the derived files in `examples/` up to date with that script. You can find more info here: https://github.com/CDCgov/trusted-intermediary/tree/main/scripts#setupupdate-examples-snapshotssh. Also, for it to work, you'd need to rename the files you added to have the `_0_initial_message.hl7` suffix

File: rs-e2e/src/main/resources/assertion_definitions.json (hunk 2 above)
Comment (saquino0827) [reply rs-e2e/src/main/resources/assertion_definitions.json]: I updated this to the suggested name `LA-Oschner OBR flattening`

File: etor/src/main/resources/transformation_definitions.json (hunk 1 above)
Comment (saquino0827) [reply etor/src/main/resources/transformation_definitions.json:175]: The new card has been written here: https://github.com/CDCgov/trusted-intermediary/issues/1641

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: K, i'll add that in a separate for the same story but that's a good idea.

Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: The sample files were created with the new script in this branch `story/1355/example-file-script`. Once this branch is merged, I'll create another PR for the new branch.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy [hunk 6]
```diff
@@ -40,8 +40,36 @@ class RemoveObservationRequestsTest extends Specification {
...
+        def sr = HapiHelper.getServiceRequest(dr)
+
+        then:
+        diagnosticReports.size() == 1
+        serviceRequests.size() == 1
+        HapiHelper.getOBR4_1Value(sr) == obr4_1
+    }
 
-    def "remove all OBRs except for the one with OBR-4.1 = '54089-8'"() {
+    def "remove all CA USCD's OBRs except for the one with OBR-4.1 = '54089-8'"() {
```
Comment (jcrichlake) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy:72]: I believe this is a typo, UCSD is the name of the hospital not USCD

Comment (saquino0827) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy:72]: Nice catch, I think it was more that I renamed it improperly since we have LA and CA OBR flattening now. I'll change it to just say `CA`

Comment (saquino0827) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy:72]: Now that I look at the the [transformation_definitions](https://github.com/CDCgov/trusted-intermediary/blob/main/etor/src/main/resources/transformation_definitions.json#L157) again, the flattening is called `ucsdOruRemoveObservationRequests` and I believe the filtering is hospital specific, so the naming is intentional

Comment (jcrichlake) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy:72]: Right but you have uSCd vs uCSd, I think the middle two letters are transposed

File: examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7 [hunk 7]
```diff
@@ -0,0 +1,76 @@
+MSH|^~\&|Natus^natus.health.state.la.us^DNS|Louisiana Office of Public Health Laboratory^2.16.840.1.114222.4.3.26.3.1^ISO|Test Department of tests^1.2.840.114350.1.13.286.2.7.2.695071^ISO|^simulated-hospital-id^L,M,N|20240101050000+0000||ORU^R01^ORU_R01|20240101050000_0001|T|2.5.1|||AL|AL|||||LRI_NG_FRN_PROFILE^^2.16.840.1.113883.9.195.3.4^ISO~LRI_NDBS_COMPONENT^^2.16.840.1.113883.9.195.3.6^ISO~LAB_PRN_Component^^2.16.840.1.113883.9.81^ISO~LAB_TO_COMPONENT^^2.16.840.1.113883.9.22^ISO|
```
Comment (basiliskus) [inline examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7:1]: All sample files in the examples folder should have 'D' in MSH-11, with the exception of files examples/Test/Automated which should have 'N'. More info here: https://github.com/CDCgov/trusted-intermediary/tree/main/examples#sample-message-files

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy (hunk 6 above)
Comment (saquino0827) [reply etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/RemoveObservationRequestsTest.groovy:72]: Ah, I missed that oops

File: examples/LA/004_LA_ORU_R01_NBS_0_initial_obr_flatten_example_2.hl7 [hunk 8]
```diff
@@ -0,0 +1,76 @@
+MSH|^~\&|Natus^natus.health.state.la.us^DNS|Louisiana Office of Public Health Laboratory^2.16.840.1.114222.4.3.26.3.1^ISO|Test Department of tests^1.2.840.114350.1.13.286.2.7.2.695071^ISO||20240101011000+0000||ORU^R01^ORU_R01|20240101011000_0002|T|2.5.1|||AL|AL|||||LRI_NG_FRN_PROFILE^^2.16.840.1.113883.9.195.3.4^ISO~LRI_NDBS_COMPONENT^^2.16.840.1.113883.9.195.3.6^ISO~LAB_PRN_Component^^2.16.840.1.113883.9.81^ISO~LAB_TO_COMPONENT^^2.16.840.1.113883.9.22^ISO|
```
Comment (basiliskus) [inline examples/LA/004_LA_ORU_R01_NBS_0_initial_obr_flatten_example_2.hl7:1]: As per comment above, MSH-11 should be updated to `D`

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (basiliskus) [inline examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: Mentioned this in a comment for another file, MSH-11 should be updated to `D` here as well

File: examples/LA/004_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_example_2.hl7 (hunk 4 above)
Comment (basiliskus) [inline examples/LA/004_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_example_2.hl7:1]: As per comment above, MSH-11 should be updated to `D`

**Review by basiliskus — COMMENTED** [review]
Left some comments about the MSH-11 expected value

File: examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7 (hunk 7 above)
Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7:1]: Okay, I already have another branch that updates the file snapshots. I'll include this change in that branch instead so I have all file updates in one location. Is that okay?

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (basiliskus) [inline examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: If you are keeping this file, it should be renamed to follow [the naming convention here](https://github.com/CDCgov/trusted-intermediary/tree/main/examples#naming-convention) (where the suffix would be `3_hl7_translation_final`)

**Review by basiliskus — COMMENTED** [review]
I noticed the files are not following the naming convention here: https://github.com/CDCgov/trusted-intermediary/tree/main/examples#naming-convention. Specifically regarding the suffixes for the flow steps, which are required for the update example script

File: examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7 (hunk 7 above)
Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7:1]: I made this change in the branch [update-snapshot-examples](https://github.com/CDCgov/trusted-intermediary/tree/updated-snapshot-examples)

File: examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7 (hunk 3 above)
Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_3_hl7_translation_final_obr_flatten_ex_1.hl7:1]: This is changed in this [branch](https://github.com/CDCgov/trusted-intermediary/tree/updated-snapshot-examples). I'll make a PR of that afterwards.

File: examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7 (hunk 7 above)
Comment (basiliskus) [reply examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7:1]: Could the files be removed from this branch and keep them in the other one to avoid confusion?

Comment (saquino0827) [reply examples/LA/003_LA_ORU_R01_NBS_0_initial_obr_flatten_example_1.hl7]: K, I removed my added `examples/LA` files. They are all in the branch mentioned above.

**Review by basiliskus — APPROVED** [review]
Thank you for addressing all my comments. Looks good!
